testshell_markdown_tutorial_crypto fails if libgpgme is not found

## Steps to Reproduce the Problem

Do not have libgpgme installed and run:
```sh
make run_all
```

## Expected Result

All tests complete sucessfully.

## Actual Result

The test `testshell_markdown_tutorial_crypto` fails, because the tool `gen-gpg-testkey` is not available.

## Suggested solution

Disable the `testshell_markdown_tutorial_crypto`, if `gen-gpg-testkey` is not generated. If this isn't possible, remove/disable the test entirely.

---

@petermax2 I assigned this to you, because you added the `gen-gpg-testkey` tool

Did you try to do a full re-build? I can not reproduce the problem. Only `testshell_markdown_path` fails (but this test usually fails in my docker installation).

Starting with the master branch (HEAD at 8c198117c3705fc79e1c9f8cd2e7c4f02022e630 ) I ran:

```sh
root@3f4532595f7c:/libelektra# mkdir build && cd build
root@3f4532595f7c:/libelektra/build# cmake -DENABLE_COVERAGE=OFF -DENABLE_OPTIMIZATIONS=OFF -DENABLE_DEBUG=ON -DENABLE_LOGGER=ON -DBUILD_STATIC=OFF -DINSTALL_SYSTEM_FILES=OFF -DPLUGINS="ALL" -DTOOLS="ALL" ..
# ... <output omitted>
-- Include Tool gen-gpg-testkey
-- Exclude tool gen-gpg-testkey because libgpgme not found
# ...
root@3f4532595f7c:/libelektra/build# make -j6
# ... 
root@3f4532595f7c:/libelektra/build# make run_all 
# ...
99% tests passed, 1 tests failed out of 224

Label Time Summary:
kdbtests    = 103.32 sec*proc (99 tests)
memleak     =  94.76 sec*proc (92 tests)

Total Test time (real) = 104.08 sec

The following tests FAILED:
	112 - testshell_markdown_path (Failed)
Errors while running CTest
tests/CMakeFiles/run_all.dir/build.make:57: recipe for target 'tests/CMakeFiles/run_all' failed
make[3]: *** [tests/CMakeFiles/run_all] Error 8
CMakeFiles/Makefile2:21466: recipe for target 'tests/CMakeFiles/run_all.dir/all' failed
make[2]: *** [tests/CMakeFiles/run_all.dir/all] Error 2
CMakeFiles/Makefile2:21473: recipe for target 'tests/CMakeFiles/run_all.dir/rule' failed
make[1]: *** [tests/CMakeFiles/run_all.dir/rule] Error 2
Makefile:5761: recipe for target 'run_all' failed
make: *** [run_all] Error 2
root@3f4532595f7c:/libelektra/build# 
```

Please provide more details on how to reproduce the problem!

--- a/cmake/ElektraCache.cmake
+++ b/cmake/ElektraCache.cmake
@@ -66,7 +66,7 @@ set (ADDED_BINDINGS
 
 remember_for_removal (TOOLS TO_REMOVE_TOOLS)
 
-set (TOOLS_LIST_DEFAULT kdb gen-gpg-testkey)
+set (TOOLS_LIST_DEFAULT kdb)
 
 if (TOOLS MATCHES "DEFAULT")
 	set (TOOLS_FORCE FORCE)

--- a/cmake/Modules/LibAddTest.cmake
+++ b/cmake/Modules/LibAddTest.cmake
@@ -118,6 +118,11 @@ function (add_msr_test NAME FILE)
 		if (tool_index GREATER -1)
 			return ()
 		endif ()
+
+		list (FIND TOOLS ${tool} tool_index)
+		if (tool_index LESS 0)
+			return ()
+		endif ()
 	endforeach ()
 
 	add_test (NAME testshell_markdown_${NAME}

--- a/cmake/Modules/LibAddTest.cmake
+++ b/cmake/Modules/LibAddTest.cmake
@@ -119,7 +119,9 @@ function (add_msr_test NAME FILE)
 			return ()
 		endif ()
 
-		list (FIND TOOLS ${tool} tool_index)
+		list (FIND TOOLS
+			   ${tool}
+			   tool_index)
 		if (tool_index LESS 0)
 			return ()
 		endif ()

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -624,6 +624,7 @@ mounted, use `kdb gen -F <plugin>:<file> elektra <parentKey> <outputName>` to lo
 ### CMake
 
 - The build system now rebuilds the [JNA binding](https://www.libelektra.org/bindings/jna) with Maven, if you change any of the Java source files of the binding. _(René Schwaiger)_
+- `testshell_markdown_tutorial_crypto` is not compiled and executed if `gen-gpg-testkey` is not part of TOOLS. _(Peter Nirschl)_
 - <<TODO>>
 - <<TODO>>
 

